core/src/main/java/me/mykindos/betterpvp/core/effects/listeners/effects/AntiHealListener.java

```diff
@@ -0,0 +1,84 @@
+package me.mykindos.betterpvp.core.effects.listeners.effects;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.core.effects.Effect;
+import me.mykindos.betterpvp.core.effects.EffectManager;
+import me.mykindos.betterpvp.core.effects.EffectTypes;
+import me.mykindos.betterpvp.core.effects.events.EffectReceiveEvent;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Particle;
+import org.bukkit.Sound;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityRegainHealthEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@BPvPListener
+@Singleton
+public class AntiHealListener implements Listener {
+
+    private final EffectManager effectManager;
+
+    @Inject
+    public AntiHealListener(EffectManager effectManager) {
+        this.effectManager = effectManager;
+    }
+
+    @EventHandler
+    public void onEntityRegainHealth(EntityRegainHealthEvent event) {
+        if (!(event.getEntity() instanceof LivingEntity ent)) return;
+        effectManager.getEffect(ent, EffectTypes.ANTI_HEAL).ifPresent(effect -> {
+            if (event.getRegainReason() == EntityRegainHealthEvent.RegainReason.SATIATED
+                    || event.getRegainReason() == EntityRegainHealthEvent.RegainReason.REGEN
+                    || event.getRegainReason() == EntityRegainHealthEvent.RegainReason.MAGIC_REGEN) {
+                if ((event.getEntity() instanceof Player player)) {
+                    player.playSound(player.getLocation(), Sound.BLOCK_GLASS_BREAK, 0.5f, 2.0f);
+                }
+                event.setCancelled(true);
+            }
+        });
+    }
+
+    @UpdateEvent
+    public void doParticles() {
+        Set<LivingEntity> affectedEntities = effectManager.getAllEntitiesWithEffects();
+
+        for (LivingEntity entity : affectedEntities) {
+            List<Effect> effects = effectManager.getEffects(entity, EffectTypes.ANTI_HEAL.getClass());
```

Comment:
just use hasEffect

Reply:
this only allows me to get vanilla effects not custom effects which might mess with some logic

Reply:
effectManager.hasEffect...